string.go

```diff
@@ -65,6 +65,7 @@ type String interface {
 	utf16Reader() utf16Reader
 	utf16RuneReader() io.RuneReader
 	utf16Runes() []rune
+	toUnicode() unicodeString
```

Comment:
The current design assumes that an ASCII string is always an `asciiString', never 'unicodeString`. There are a couple of optimisations based on this assumption (like the equality operator for example). Even though this is only used for regexp, having this method on the interface would tempt someone to misuse it at some point.

A better way would be either to have `utf16() []uint16` method instead, or, use devirtualizeString.